logbook-api/src/main/java/org/zalando/logbook/HttpHeaders.java

```diff
@@ -1,7 +1,7 @@
 package org.zalando.logbook;
 
-import javax.annotation.CheckReturnValue;
```

Comment:
`CheckReturnValue` can be imported by using `com.github.spotbugs:spotbugs-annotations`; see:

https://github.com/findbugsproject/findbugs/blob/master/findbugs/src/java/edu/umd/cs/findbugs/annotations/CheckReturnValue.java

Reply:
After some research, I was also going to suggest JSpecify. 

I think the proper annotation would be `@NullMarked`.

I guess optional is fine for our package compilation, but do our clients benefit from it if they forget to add it?

Reply:
To my knowledge, `@NullMarked` is a per-package annotation; it automatically applies to all parameters and return values in that package.

My concern,as stated in previous comment, was about making JSpecify optional.

Reply:
The dilemma I have with making it not optional is: should we make it compile or provided? Compile would bring the transitive dependency to deal with to the clients, but provided would force them to add one, which is somehow feels even worse.